packages/db/src/models/granule.ts

```diff
@@ -37,6 +86,30 @@ export default class GranulePgModel extends BasePgModel<PostgresGranule, Postgre
       .merge()
       .returning('cumulus_id');
   }
+
+  async upsertWithExecutionHistory(
+    knexOrTrx: Knex | Knex.Transaction,
+    granule: PostgresGranule,
+    executionCumulusId: number,
+    granuleExecutionHistoryPgModel = new GranuleExecutionHistoryPgModel()
+  ): Promise<number[]> {
+    const [granuleCumulusId] = await this.upsert(knexOrTrx, granule, executionCumulusId);
+    // granuleCumulusId could be undefined if the upsert affected no rows due to its
+    // conditional logic. In that case, we assume that the executino history for the
+    // granule was already written and return early. Execution history cannot be written
+    // without granuleCumulusId regardless.
+    if (!granuleCumulusId) {
+      return [];
+    }
```

Comment:
Does this logic seem sane?

Reply:
I'm not sure it's insane, but  this logic feels like something that should be handled out of this model, or if we want to tie the join table inserts, just *assume* that we're going to upsert the row into the granules-executions join table in the `upsert` method.

Reply:
If we didn't do the logic for writing to the join table from this model, where would we do it? Everywhere that we call `granulePgModel.upsert()`? I suppose there is merit to that, though it felt like since granules-executions are now intrinsically linked and this is all logic for talking to the database, it made sense to have the granule PG model handled it.

I'm also not sure I understand what assuming would mean in this circumstance.

Reply:
TODO: this whole method will also move to a granule lib/helper putting togher granule writes and granule/execution writes